docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+                       requestContext: OidcRequestContext<OidcTenantConfig>?): Uni<OidcTenantConfig>? {
+
+    // #0) Check sanity
+    //
+    if (routingContext == null) {
+
+      log.warn("Received null RoutingContext. Cannot determine realm; reverting to default tenant.")
+      return Uni.createFrom().nullItem()
+    }
+
+    // #1) Find the tenant identifier/realm, from the path.
+    //
+    val contextPath = routingContext.request().path()
+    val realm = getRealmFrom(contextPath)
+
+    if (realm == null) {
+
+      log.warn("== No Realm derived from request path [$contextPath]")
+      return Uni.createFrom().nullItem()
+    } else {
+      log.debug("== Realm [$realm] found from request path [$contextPath]")
+    }
+
+    // #2) Fallback to null result if the tenant identifier was not recognized
+    //
+    return when (val config = REALM_2_OIDC_CONFIG[realm]) {
+      null -> Uni.createFrom().nullItem()
+      else -> Uni.createFrom().item(config)
+    }
+  }
+
+  companion object {
+
+    @JvmStatic
+    internal val REALM_2_OIDC_CONFIG: SortedMap<String, Supplier<OidcTenantConfig>> = TreeMap()
+
+    @JvmStatic
+    private val log = LoggerFactory.getLogger(OidcConfigurationResolver::class.java)
+  }
+}
+----
+
+This implementation concludes steps `1` in Figure 2 above. Quarkus is now aware of the tenant identifier/realm, and has resolved the configuration for the OIDC provider which authenticated the calling user.
+
+=== 2. Retrieve injected IdToken and user information
+
+Quarkus provides means to inject the current user's information into the multi-tenant application's resources. It is often practical to access these properties in a standard way within your RESTful resource implementations, such as within a common RESTful resource superclass.
+
+There are several options to inject OIDC-based user information from Quarkus's security framework into your resources, 3 of which are illustrated in the code snippet below:
+
+. `Inject the AccessToken` typecast to a `JsonWebToken`, which holds information from OIDC provider with simple access methods. This is a `@RequestScoped` CDI bean, which is automatically provided by the `quarkus-oidc` extension. As discussed above, the access token holds user authorization information, so it is the best option to use when validating that the user is authorized to perform operations within the multi-tenant service.
+. _(Preferred):_ `Inject a custom JsonWebToken wrapper` exemplified by a TokenBasedCallerInfo below. Depending on your corporate/business requirements
+. _(Discouraged):_ `Inject the IdToken` typecast to a `JsonWebToken`, which holds information from the IdToken generated by the OIDC provider. This is a `@RequestScoped` CDI bean, which is automatically provided by the `quarkus-oidc` extension. You need to qualify the injection point with `@IdToken` - but OAuth2 authentication information is not necessarily part of the IdToken as discussed above.
```

Comment:
As I said above, I think all the text related to the treatment of tokens should be adapted to the bearer and code guides. Also I'd not qualify it as `Discouraged` - for most users not working with Oauth2 but certified OIDC providers it will be a real ID token.